lib/drivers/tandemDriver.js

```diff
@@ -1288,15 +1288,13 @@ module.exports = function (config) {
       targetSchedules[scheduleName] = targetSchedule;
     });
 
-    var postsettings = cfg.builder.makePumpSettings()
+    var postsettings = cfg.builder.makePumpSettingsAllObjects()
       .with_activeSchedule(activeName)
       .with_units({ carb: 'grams', bg: 'mg/dL' })
       .with_basalSchedules(basalSchedules)
-      // TODO: remove filtering on activeName so all profiles are included
-      // and will need to add & use with_carbRatios, with_insulinSensitivities, with_bgTargets
-      .with_carbRatio(carbSchedules[activeName])
-      .with_insulinSensitivity(sensitivitySchedules[activeName])
-      .with_bgTarget(targetSchedules[activeName])
+      .with_carbRatios(carbSchedules)
+      .with_insulinSensitivities(sensitivitySchedules)
+      .with_bgTargets(targetSchedules)
       .with_time(sundial.applyTimezone(data.log_records[data.log_records.length - 1].jsDate, cfg.timezone).toISOString())
```

Comment:
I'm now using the "device time in settings packet at time of upload" timestamp, and added an explanation in the comments.